dashboard/app/models/user.rb

```diff
@@ -835,8 +835,8 @@ def User.track_proficiency(user_id, script_id, level_id)
 
   # returns whether a new level has been completed and asynchronously enqueues an operation
   # to update the level progress.
-  def track_level_progress_async(script_level:, new_result:, submitted:, level_source_id:)
-    level_id = script_level.level_id
+  def track_level_progress_async(script_level:, new_result:, submitted:, level_source_id:, level:)
+    level_id = level.try(:id) || script_level.level_id
```

Comment:
@balderdash - this will likely interact with your feature. I discussed this with @laurelfan and it seems like it makes sense even for your scenario to store the progress on the underlying level (not the "parent" level in the script), but we'll probably need to make a corresponding change when reading the script progress for student or teacher progress such that we're aware of this. Or, for your scenario, we could create a `UserLevel` for both the parent level in the script and the actual level that they submitted progress for. Thoughts?